test/Microsoft.Extensions.Primitives.Tests/StringSegmentTest.cs

```diff
@@ -20,6 +20,41 @@ public void StringSegment_StringCtor_AllowsNullBuffers()
             Assert.Equal(0, segment.Length);
         }
 
+        [Fact]
+        public void StringSegmentConstructor_NullBuffer_Throws()
+        {
+            // Arrange, Act and Assert
+            var exception = Assert.Throws<ArgumentNullException>(() => new StringSegment(null, 0, 0));
+            Assert.Contains("buffer", exception.Message);
```

Comment:
Could you change this to `Assert.Equal("buffer", expection.ParameterName);` and similarly for the other ones. `Contains` is usually not a very good way to verify these.